src/main/java/com/powsybl/openloadflow/sa/AbstractSecurityAnalysis.java

```diff
@@ -89,23 +90,32 @@ public CompletableFuture<SecurityAnalysisReport> run(String workingVariantId, Se
     abstract SecurityAnalysisReport runSync(String workingVariantId, SecurityAnalysisParameters securityAnalysisParameters, ContingenciesProvider contingenciesProvider,
                                             ComputationManager computationManager, List<OperatorStrategy> operatorStrategies, List<Action> actions);
 
-    public static PostContingencyComputationStatus postContingencyStatusFromNRStatus(NewtonRaphsonStatus status) {
-        return switch (status) {
-            case CONVERGED -> PostContingencyComputationStatus.CONVERGED;
-            case MAX_ITERATION_REACHED -> PostContingencyComputationStatus.MAX_ITERATION_REACHED;
-            case SOLVER_FAILED -> PostContingencyComputationStatus.SOLVER_FAILED;
-            case NO_CALCULATION -> PostContingencyComputationStatus.NO_IMPACT;
-            case UNREALISTIC_STATE -> PostContingencyComputationStatus.FAILED;
-        };
+    public static PostContingencyComputationStatus postContingencyStatusFromAcLoadFlowResult(AcLoadFlowResult result) {
+        if (result.getOuterLoopStatus() == OuterLoopStatus.UNSTABLE) {
+            return PostContingencyComputationStatus.MAX_ITERATION_REACHED;
+        } else {
+            return switch (result.getNewtonRaphsonStatus()) {
+                case CONVERGED -> PostContingencyComputationStatus.CONVERGED;
+                case MAX_ITERATION_REACHED -> PostContingencyComputationStatus.MAX_ITERATION_REACHED;
+                case SOLVER_FAILED -> PostContingencyComputationStatus.SOLVER_FAILED;
+                case NO_CALCULATION -> PostContingencyComputationStatus.NO_IMPACT;
+                case UNREALISTIC_STATE -> PostContingencyComputationStatus.FAILED;
+            };
+        }
     }
 
-    public static LoadFlowResult.ComponentResult.Status loadFlowResultStatusFromNRStatus(NewtonRaphsonStatus status) {
-        return switch (status) {
-            case CONVERGED -> LoadFlowResult.ComponentResult.Status.CONVERGED;
-            case MAX_ITERATION_REACHED -> LoadFlowResult.ComponentResult.Status.MAX_ITERATION_REACHED;
-            case SOLVER_FAILED -> LoadFlowResult.ComponentResult.Status.SOLVER_FAILED;
-            case NO_CALCULATION, UNREALISTIC_STATE -> LoadFlowResult.ComponentResult.Status.FAILED;
-        };
+    public static LoadFlowResult.ComponentResult.Status componentResultStatusFromAcLoadFlowResult(AcLoadFlowResult result) {
+        // FIXME duplicates com.powsybl.openloadflow.OpenLoadFlowProvider.convertStatus
```

Comment:
@obrix @geofjamg would it be OK to just turn OpenLoadFlowProvider.convertStatus public ?

Reply:
I would move it to AcLoadFlowResult as a static utility method.

Reply:
thanks, done